src/extractor/extraction_containers.cpp

```diff
@@ -625,6 +635,40 @@ void ExtractionContainers::WriteNodes(storage::io::FileWriter &file_out) const
         log << "ok, after " << TIMER_SEC(write_nodes) << "s";
     }
 
+    {
+        util::UnbufferedLog log;
+        log << "Writing barrier nodes     ... ";
+        TIMER_START(write_nodes);
+        std::vector<NodeID> internal_barrier_nodes;
+        for (const auto id : barrier_nodes)
+        {
+            auto iter = external_to_internal_node_id_map.find(id);
+            if (iter != external_to_internal_node_id_map.end())
```

Comment:
Hmm, is this test necessary?  Every `id` should exist in the map, we can probably just `BOOST_ASSERT()` the comparison here and assume `iter->second` is good.